packages/vite/src/node/plugins/css.ts

```diff
@@ -792,7 +788,30 @@ export function cssPostPlugin(config: ResolvedConfig): Plugin {
         })
       }
 
-      let extractedCss = outputToExtractedCSSMap.get(opts)
+      function extractCss() {
+        let css = ''
+        const collected = new Set<OutputChunk>()
+        const prelimaryNameToChunkMap = new Map(
+          Object.values(bundle)
+            .filter((chunk): chunk is OutputChunk => chunk.type === 'chunk')
+            .map((chunk) => [chunk.preliminaryFileName, chunk]),
+        )
+
+        function collect(fileName: string) {
+          const chunk = bundle[fileName]
+          if (!chunk || chunk.type !== 'chunk' || collected.has(chunk)) return
+          collected.add(chunk)
+
+          chunk.imports.forEach(collect)
+          css += chunkCSSMap.get(chunk.preliminaryFileName) ?? ''
+        }
+
+        for (const chunkName of chunkCSSMap.keys())
+          collect(prelimaryNameToChunkMap.get(chunkName)?.fileName ?? '')
```

Comment:
Regarding this line, I was wondering if the order of `Object.keys(bundle)` corresponds to how rollup traverses during the `renderChunk` phase. If it does then we can eliminate the `preliminaryNameToChunkMap` here. Though in my small project for testing purpose it seems true, I can't tell whether it holds generally.

Reply:
Let's go with this implementation as I don't think rollup guarantees that.